docs/sources/visualizations/histogram-draft.md

```diff
@@ -0,0 +1,107 @@
+| `smaller than 1MB`            |   `uploaded_image_bytes_bucket{le="1048576"} / ignoring (le) uploaded_image_bytes_count`              |       number of files less than (or equal to) 1MB, total number of files       | The total count of files. Total count for a histogram can be found using either `uploaded_image_bytes_count` or `uploaded_image_bytes_bucket{le="+Inf"}`. Divide the number of files smaller than 1MB by the total number of files to get a ratio between the two. Since the normal way of displaying ratios is as percentages, set the unit to `Percent (0.0-1.0)`.|
+| `larger than 1MB`             |   `uploaded_image_bytes_count - ignoring(le) uploaded_image_bytes_bucket{le="1048576"}`               |       number of smaller files, number of total files       | Subtract the number of smaller files from the number of total files to get the number of larger files.  |  
+| `between 256KB and 1MB`       |   `uploaded_image_bytes_bucket{le="1048576"} - ignoring(le) uploaded_image_bytes_bucket{le="262144"}` |       number or files between any two bucket boundaries       | Using the same logic as for the previous query, get the number of files between any two bucket boundaries by subtracting the smaller boundary from the larger.  |
+| `approximation using PromQL`  |   `histogram_quantile(0.75, uploaded_image_bytes_bucket)`                                             |       ratio, buckets       | The function takes a ratio and the histogram’s buckets as input and returns an approximation of the value at the point of the ratio’s quantile. |
+
+You can use one of the following **panels** to visualize their query:
+
+- **Stat.** Shows the summary of a single series; displays only one metric per panel
+- **Gauge.** Also known as a speedometer; best used when the data had defined boundary limits to warn users when they are falling under the normal range
+- **Graph.** Time charts that display data points over a time axis allowing users to overlap metrics to compare them overtime; easily modified and good for tracking outliers, state changes, or triggers
+
+This sample query is best represented using a stat if users want to see files they currently have as opposed to over time (which is better shown as a graph). A gauge would not be feasible without a defined range.
+
+## Using PromQL to approximate
+
+We cannot get an accurate value of what size is a quarter of the files smaller than, but we can get an approximation by using PromQL’s `histogram_quantile` function.
```

Comment:
```suggestion
The number of files uploaded that are less than (or equal to) 1MB is really the base question for a Prometheus histogram. The answer is stored in the time series database already and we don’t need to use either functions or arithmetic to answer the question. Adding up the sum of buckets manually would be both error-prone and complicated. We can assume we’d have a helper function in case we’d have to do that, the operation would also mean that even simple queries like this one would have to look at a substantial number of unique time series rather than just one. We can thus use PromQL for the same visualization by asking a few more questions:
- What proportion of the files are smaller than 1MB?
- How many files are larger than 1MB?
- How many files are in the bucket between 256KB and 1MB?
- What size is a quarter of the files smaller than?

We cannot get an accurate value of what size is a quarter of the files smaller than, but we can get an approximation by using PromQL’s `histogram_quantile` function. The function takes a ratio and the histogram’s buckets as input and returns an approximation of the value at the point of the ratio’s quantile. (that is, if 1 is the largest file and 0 is the smallest file, how big would file 0.75 be?)


```